### PR TITLE
ASU-1606: HASO apartment revaluation

### PR DESCRIPTION
**Pull request by voneiden**

### CODE REVIEW

File: src/components/revaluation/ApartmentRevaluationModal.module.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+form {
```
Comment (juhakujala) [inline src/components/revaluation/ApartmentRevaluationModal.module.scss:1]: I don't think that this actually targets any `<form>` components as it should be referenced in the `.tsx` file as a `className`

File: src/components/revaluation/ReservationRevaluationInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { Button, IconAlertCircle, IconPen } from 'hds-react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+
+import { showApartmentRevaluationModal } from '../../redux/features/apartmentRevaluationModalSlice';
+import { ApartmentReservationWithCustomer } from '../../types';
+import formatDateTime from '../../utils/formatDateTime';
+
+import '../costindex/CostIndexSingleTable.module.scss';
```
Comment (juhakujala) [inline src/components/revaluation/ReservationRevaluationInfo.tsx:10]: Is this actually relevant?

File: src/components/revaluation/ReservationRevaluationInfo.module.scss [hunk 3]
```diff
@@ -0,0 +1,42 @@
+.tooltip {
```
Comment (juhakujala) [inline src/components/revaluation/ReservationRevaluationInfo.module.scss:1]: Was there a tooltip for this component?

File: src/i18n/en.json [hunk 4]
```diff
@@ -11,6 +11,26 @@
...
+  "components.revaluation.ApartmentRevaluationForm.forCustomer": "Right of occupancy payment customer",
+  "components.revaluation.ApartmentRevaluationForm.originalRightOfOccupancyPayment": "Original right of occupancy payment (€)",
+  "components.revaluation.ApartmentRevaluationForm.costIndexStartDate": "1st original payment due date",
+  "components.revaluation.ApartmentRevaluationForm.costIndexValueForStartDate": "Cost index on 1st original payment due date",
+  "components.revaluation.ApartmentRevaluationForm.alterationWork": "Alteration work (€)",
+  "components.revaluation.ApartmentRevaluationForm.costIndexEndDate": "Contract end date",
+  "components.revaluation.ApartmentRevaluationForm.costIndexValueForEndDate": "Contract end date cost index",
+  "components.revaluation.ApartmentRevaluationForm.adjustedRightOfOccupancyPayment": "Revalued cost index",
+  "components.revaluation.ApartmentRevaluationForm.submit": "Save",
+  "components.revaluation.ApartmentRevaluationForm.cancel": "Camceö",
```
Comment (juhakujala) [inline src/i18n/en.json:31]: Oops 😄

**Review by juhakujala — CHANGES_REQUESTED** [review]
Good job overall 👍 There's some minor things that could be fixed before merging

File: src/components/revaluation/ReservationRevaluationInfo.module.scss (hunk 3 above)
Comment (voneiden) [reply src/components/revaluation/ReservationRevaluationInfo.module.scss:1]: Lets say "there was an attempt", but it really didn't work in the list view as expected. So yeah, leftover trash.

File: src/components/revaluation/ApartmentRevaluationModal.module.scss (hunk 1 above)
Comment (voneiden) [reply src/components/revaluation/ApartmentRevaluationModal.module.scss:1]: It does have an effect, but good catch I'll change it to a class reference so it doesn't cause any unintended side effects to other forms.

File: src/components/revaluation/ApartmentRevaluationFormContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+  ApartmentReservationCustomer,
+  ApartmentReservationWithCustomer,
+  ApartmentRevaluation,
+  CostIndex,
+} from '../../types';
+import Container from '../common/container/Container';
+import Spinner from '../common/spinner/Spinner';
+import ApartmentRevaluationForm from './ApartmentRevaluationForm';
+
+const T_PATH = 'components.revaluation.ApartmentRevaluationModal';
```
[the remark this answers is not in the document]
Comment (voneiden) [reply src/components/revaluation/ApartmentRevaluationFormContainer.tsx:23]: Was intentional, but yeah, inconsistent.  Renaming

File: src/components/reservations/ReservationCancelModal.tsx [hunk 6]
```diff
@@ -44,10 +50,9 @@ const ReservationCancelModal = (): JSX.Element | null => {
     if (!postReservationCancelLoading) {
       setIsLoading(true);
 
-      // Project uuid, customer id and Apartment uuid is used to invalidate cached data after cancelling a reservation
+      // Project uuid and customer id is used to invalidate cached data after cancelling a reservation
       const projectId = reservationCancelModal.content?.projectId || '';
       const customerId = reservationCancelModal.content?.customer.id || 0;
-      const apartmentId = reservationCancelModal.content?.apartmentId || '';
```
[the remark this answers is not in the document]
Comment (voneiden) [reply src/components/reservations/ReservationCancelModal.tsx:50]: Oh yeah good spotting, this was indeed a leftover change from development. I'll clean up the whole file, only the part which opens the revaluation modal is necessary.

**Review by juhakujala — APPROVED** [review]

**Pull request merged**